src/Microsoft.VisualStudio.ProjectSystem.VisualBasic.VS/ProjectSystem/VS/Properties/MapDynamicEnumValuesProvider.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.ProjectSystem.Properties;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Properties
+{
+    /// <summary>
+    /// This Dynamic Enum Value Provider maps enum based msbuild property value
+    /// to values that we want to display in the UI and also map the value,we 
+    /// obtain from the UI to msbuild complaint values for persistence
+    /// </summary>
+    internal class MapDynamicEnumValuesProvider : IDynamicEnumValuesGenerator
+    {
+        private readonly IDictionary<string, IEnumValue> _getValueMap;
```

Comment:
> from the two providers you have the Values are same between both maps

The values are the same in the case of `WarningLevelEnumProvider` but in the case of `OptionStrictEnumProvider`, the values are not the same. But I see what you are saying. Let me refactor this a little.